MovieQuiz/Presentation/MovieQuizPresenter.swift

```diff
@@ -0,0 +1,162 @@
+//
+//  MovieQuizPresenter.swift
+//  MovieQuiz
+//
+//  Created by Роман Бойко on 11/14/22.
+//
+
+import Foundation
+
+import UIKit
+
+final class MovieQuizPresenter: QuestionFactoryDelegate {
+    
+    // Сервис сбора статистики
+    private let statisticService: StatisticServiceImplementation!
+
+    // Переменная для подсчёта колличества верных ответов
+    private var correctAnswers: Int = 0
+    
+    // Экземпляр фабрики вопросов
+    private var questionFactory: QuestionFactoryProtocol?
+    
+    // Переменная индекса текущего вопроса
+    private var currentQuestionIndex: Int = 0
+    
+    // Общее колличество вопросов
+    private let questionsAmount: Int = 10
+    
+    // Текущий вопрос, который видит пользователь
+    private var currentQuestion: QuizQuestion?
+    
+    // Слабая ссылка на ViewController
+    private weak var viewController: MovieQuizViewControllerProtocol?
+    
+    init(viewController: MovieQuizViewControllerProtocol) {
+        self.viewController = viewController
+        statisticService = StatisticServiceImplementation()
+        questionFactory = QuestionFactory(moviesLoader: MoviesLoader(), delegate: self)
+        questionFactory?.loadData()
+        viewController.showLoadingIndicator()
+    }
+    
+    // MARK: - Methods
+    // Функция для кнопки "Да"
+    func yesButtonClicked() {
+        didAnswer(isYes: true)
+    }
+    
+    // Функция для кнопки "Нет"
+    func noButtonClicked() {
+        didAnswer(isYes: false)
+    }
+    
+    // Функция сравнения ответа пользователя и правильного ответа
+    func didAnswer (isYes: Bool) {
+        guard let currentQuestion = currentQuestion else {
+            return
+        }
+        let isCorrect = (isYes == currentQuestion.correctAnswer)
+        if isCorrect {
+            correctAnswers += 1
+        }
+        
+        self.showAnswerResult(isCorrect: isCorrect)
+    }
+    
+    // Функция преобразования вопроса в вью модель
+    func convert(model: QuizQuestion) -> QuizStepViewModel {
+        return QuizStepViewModel(
+            image: UIImage(data: model.image) ?? UIImage(),
+            question: model.text,
+            questionNumber: String("\(currentQuestionIndex + 1)/\(questionsAmount)")
+        )
+    }
+    
+    // Фунуция для определения последнего вопроса
+    func isLastQuestion () -> Bool {
```

Comment:
Кажется, здесь стоит проверить функции на необходимость выставить модификаторы доступа
```suggestion
    private func isLastQuestion() -> Bool {
```